linera-views/src/key_value_store_view.rs

```diff
@@ -741,18 +722,16 @@ where
                 result.push(value);
             } else {
                 result.push(None);
-                if !contains_key(&self.deleted_prefixes, &index) {
+                if !self.deletion_set.contains_prefix_of(&index) {
                     missed_indices.push(i);
                     let key = self.context.base_tag_index(KeyTag::Index as u8, &index);
                     vector_query.push(key);
                 }
             }
         }
-        if !self.delete_storage_first {
```

Comment:
Also this one?

Reply:
If `self.delete_storage_first` then the vector si empty which would produce an empty vector. I do not think that there is an optimization opportunity here.